File: src/test/java/com/natu/ftax/IDgenerator/infrastructure/SequentialIdGenerator.java
package com.natu.ftax.IDgenerator.infrastructure;

import com.natu.ftax.IDgenerator.domain.IdGenerator;

import java.util.concurrent.atomic.AtomicLong;

public class SequentialIdGenerator implements IdGenerator {
    private final AtomicLong counter = new AtomicLong(0);

    @Override
    public String generate() {
        return Long.toString(counter.incrementAndGet());
    }
}

File: src/test/kotlin/com/natu/ftax/IDgenerator/infrastructure/SequentialIdGenerator.kt
package com.natu.ftax.IDgenerator.infrastructure

import com.natu.ftax.IDgenerator.domain.IdGenerator
import java.util.concurrent.atomic.AtomicLong

class SequentialIdGenerator : IdGenerator {
    private val counter = AtomicLong(0)

    override fun generate(): String {
        return counter.incrementAndGet().toString()
    }
}